src/interfaces/services/deploy.ts

```diff
@@ -0,0 +1,23 @@
+export abstract class IDeployService {
+	abstract getAddress(
+		bytecode: string,
+		setExpectedAddress: (value: `0x${string}`) => void
+	): Promise<void>;
+
+	abstract sign(
```

Comment:
Should sign be part of the `DeployService`, or should there be a `SignatureService`?

Reply:
I think it should be part of the deploy service, since the message hash that the user is signing comes from the PredictiveDeployer contract.